tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -4167,8 +4217,13 @@ class HLSLExternalSource : public ExternalSemaSource {
     for (unsigned int i = 0; i < tableSize; i++) {
       const HLSL_INTRINSIC* pIntrinsic = &table[i];
 
+      // For intrinsics with variable number of arguments (variadic functions),
+      // we should not compare the number of arguments.
+      // For such cases, the last argument's type is INTRIN_TEMPLATE_VARARGS.
+      const bool isVariadicFn = IsVariadicIntrinsicFunction(pIntrinsic);
```

Comment:
This comment looks like perhaps it predated separateing out the isVariadicIntrinsicFunction to its own thing. It might be more helpful in that function.